roofit/roofitcore/src/RooDataHist.cxx

```diff
@@ -1263,8 +1263,18 @@ double RooDataHist::weightInterpolated(const RooArgSet& bin, int intOrder, bool
     auto idxMultY = _idxMult[varInfo.realVarIdx2];
     auto offsetIdx = centralIdx - idxMultY * ybinC;
 
-    std::vector<double> yarr(intOrder+1);
-    std::vector<double> xarr(intOrder+1);
+    // Optimization for small interpolation orders (intOrder <= 2) that avoids
```

Comment:
Is this really making a difference the vector allocation ? 
Since the function is not const, why not using a buffer member of the class that is re-used every time ?